lizrd/train/checkpoints_manager.py

```diff
@@ -0,0 +1,311 @@
+    model_checkpoint_path: str,
+    job_id: str,
+    timestamp: str,
+    metadata: Optional[str] = None,
+):
+    return {
+        CHECKPOINT_STATUS: CHECKPOINT_STATUS_PENDING,
+        MODEL_CHECKPOINT: model_checkpoint_path,
+        CHECKPOINT_RUNNING_JOB_ID: None,
+        CHECKPOINT_START_TIMESTAMP: None,
+        CHECKPOINT_CREATOR_JOB_ID: job_id,
+        CHECKPOINT_CREATE_TIMESTAMP: timestamp,
+        CHECKPOINT_STOP_TIMESTAMP: None,
+        FINAL_MODEL_CHECKPOINT: None,
+        CHECKPOINT_METADATA_TAG: metadata,
+    }
+
+
+def run_manager_checkpoint(job_id: str, timestamp: str, checkpoint: dict):
+    assert checkpoint[CHECKPOINT_STATUS] == CHECKPOINT_STATUS_PENDING
+    return {
+        CHECKPOINT_STATUS: CHECKPOINT_STATUS_RUNNING,
+        MODEL_CHECKPOINT: checkpoint[MODEL_CHECKPOINT],
+        CHECKPOINT_RUNNING_JOB_ID: job_id,
+        CHECKPOINT_START_TIMESTAMP: timestamp,
+        CHECKPOINT_CREATOR_JOB_ID: checkpoint[CHECKPOINT_CREATOR_JOB_ID],
+        CHECKPOINT_CREATE_TIMESTAMP: checkpoint[CHECKPOINT_CREATE_TIMESTAMP],
+        CHECKPOINT_STOP_TIMESTAMP: None,
+        FINAL_MODEL_CHECKPOINT: None,
+        CHECKPOINT_METADATA_TAG: checkpoint[CHECKPOINT_METADATA_TAG],
+    }
+
+
+def finish_manager_checkpoint(
+    model_checkpoint_path: str, timestamp: str, checkpoint: dict
+):
+    assert checkpoint[CHECKPOINT_STATUS] == CHECKPOINT_STATUS_RUNNING
+    return {
+        CHECKPOINT_STATUS: CHECKPOINT_STATUS_FINISHED,
+        MODEL_CHECKPOINT: checkpoint[MODEL_CHECKPOINT],
+        CHECKPOINT_RUNNING_JOB_ID: checkpoint[CHECKPOINT_RUNNING_JOB_ID],
+        CHECKPOINT_START_TIMESTAMP: checkpoint[CHECKPOINT_START_TIMESTAMP],
+        CHECKPOINT_CREATOR_JOB_ID: checkpoint[CHECKPOINT_CREATOR_JOB_ID],
+        CHECKPOINT_CREATE_TIMESTAMP: checkpoint[CHECKPOINT_CREATE_TIMESTAMP],
+        CHECKPOINT_STOP_TIMESTAMP: timestamp,
+        FINAL_MODEL_CHECKPOINT: model_checkpoint_path,
+        CHECKPOINT_METADATA_TAG: checkpoint[CHECKPOINT_METADATA_TAG],
+    }
+
+
+def release_checkpoint_manager(
+    manager_content, job_id, model_checkpoint_path, timestamp
+):
+    new_manager_content = deepcopy(manager_content)
+    have_released = False
+    for i, checkpoint in enumerate(manager_content[CHECKPOINTS_TAG]):
+        if (
+            checkpoint[CHECKPOINT_RUNNING_JOB_ID] == job_id
+            and checkpoint[CHECKPOINT_STATUS] == CHECKPOINT_STATUS_RUNNING
+        ):
+            new_manager_content[CHECKPOINTS_TAG][i] = finish_manager_checkpoint(
+                model_checkpoint_path, timestamp, checkpoint
+            )
+            have_released = True
+    assert have_released == True
+    return new_manager_content
+
+
+def __overwrite_manager(new_content, f):
+    f.seek(0)
+    f.truncate()
+    json.dump(new_content, f, indent=4)
+
+
+def __get_manager_timestamp():
+    return datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3]
+
+
+def start_job_manager_assesment(job_id: str, is_logging_process):
```

Comment:
The docstring is misleading (the return types are not None | str). Let's add types and fix that.

Reply:
also _assessment_ is written with two s's.

Reply:
done